core/types/receipt.go

```diff
@@ -438,5 +458,26 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 			logIndex++
 		}
 	}
+	if config.Optimism != nil && len(txs) >= 2 { // need at least an info tx and a non-info tx
```

Comment:
What do you think of hiding this logic in a function and calling that function here

Reply:
it's only used in this one place, but if you like it better we can do that

Reply:
yea, I'd like a function to get the values & another to compute the fees.

Reply:
Hmm, it does need to write to the receipts, and by splitting it up in a function we need to support that writing interface, where receipts might not match the transactions etc. Those additional error edge cases do not seem worth it.